modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -3886,7 +3898,14 @@ def n_ary_op(
                 1,
                 new_right_frames,
                 join_type,
-                sort=True,
+                sort=(
+                    not all(
+                        self.get_axis(1).equals(right.get_axis(1))
```

Comment:
would it make sense to move this logic to def _copartition and do this there if sort is None?

Reply:
makes sense, done